scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,382 @@
+    except IOError:
+        print("Couldn't read /etc/os-release")
+    return distribution_id, version_id
+
+
+def is_non_eligible_releases(release):
+    print("Exit if not RHEL 7.9 or 8.4")
+    major_version, minor = release.split(".") if release is not None else (None, None)
+    version_str = major_version + "." + minor
+    eligible_releases = ["7.9", "8.4"]
+    return release is None or version_str not in eligible_releases
+
+
+# Code taken from
+# https://github.com/oamg/convert2rhel/blob/v1.4.1/convert2rhel/utils.py#L345
+# and modified to adapt the needs of the tools that are being executed in this
+# script.
+def run_subprocess(cmd, print_cmd=True, env=None, wait=True):
+    """
+    Call the passed command and optionally log the called command
+    (print_cmd=True) and environment variables in form of dictionary(env=None).
+    Switching off printing the command can be useful in case it contains a
+    password in plain text.
+
+    The cmd is specified as a list starting with the command and followed by a
+    list of arguments. Example: ["yum", "install", "<package>"]
+    """
+    if isinstance(cmd, str):
+        raise TypeError("cmd should be a list, not a str")
+
+    if print_cmd:
+        print("Calling command '%s'" % " ".join(cmd))
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1, env=env
+    )
+
+    output = ""
+
+    if not wait:
+        return output, None
+
+    for line in iter(process.stdout.readline, b""):
+        line = line.decode("utf8")
+        output += line
+
+    # Call wait() to wait for the process to terminate so that we can
+    # get the return code.
+    process.wait()
+
+    return output, process.returncode
+
+
+def check_if_package_installed(pkg_name):
+    _, return_code = run_subprocess(["rpm", "-q", pkg_name])
+    return return_code == 0
+
+
+def setup_leapp(command, rhui_pkgs):
+    print("Installing leapp ...")
+    output, returncode = run_subprocess(command)
+    if returncode:
+        print(
+            "Installation of leapp failed with code '%s' and output: %s\n"
+            % (returncode, output)
+        )
+        raise ProcessError(
+            message="Installation of leapp failed with code '%s'." % returncode
+        )
+
+    print("Check installed rhui packages ...")
+    for pkg in rhui_pkgs:
+        if check_if_package_installed(pkg["src_pkg"]):
+            pkg["installed"] = True
+    return [pkg for pkg in rhui_pkgs if pkg.get("installed", False)]
+
+
+def should_use_no_rhsm_check(rhui_installed, command):
+    print("Checking if subscription manager and repositories are available ...")
+    rhsm_repo_check_fail = True
+    _, rhsm_installed_check = run_subprocess(["which", "subscription-manager"])
+    if rhsm_installed_check == 0:
+        rhsm_repo_check, _ = run_subprocess(
+            ["subscription-manager", "repos", "--list-enabled"]
```

Comment:
Use the full path to the subscription-manager binary: 

```suggestion
            ["/usr/sbin/subscription-manager", "repos", "--list-enabled"]
```